lib/internal/modules/cjs/loader.js

```diff
@@ -84,6 +84,25 @@ const relativeResolveCache = Object.create(null);
 
 let requireDepth = 0;
 let statCache = null;
+
+function enrichCJSError(err) {
+  const stack = err.stack.split('\n');
+
+  const lineWithErr = stack[1];
+
+  /*
+    The regular expression below targets the most common import statement
+    usage, however, some cases are not matching, cases like import statement
+    after a comment block and/or after a variable definition.
+  */
+  if (err.message.startsWith('Unexpected token export') ||
+    (/^\s*import(?=[ {'"*])\s*(?![ (])/).test(lineWithErr)) {
+    err.stack = 'Note: To load an ES module using --experimental-modules ' +
```

Comment:
Should we call out the experimental flag in this error message?

```suggestion
    err.stack = 'Note: To load an ES module using --experimental-modules ' +
    err.stack = 'Note: To load an ES module from a .js file ' +
```

Reply:
> Should we call out the experimental flag in this error message?

I believe it's better to keep like it is when ECMAScript Modules change stability to the stable phase I'll update the note. Less confusion for users until that happens.

Reply:
My point here is that it's less connected to `--experimental-modules` and more to the way it is being authored and/or imported. Right now the message reads to me as "you are using --experimental-modules, please set type in package.json or use mjs". Which doesn't seem to catch the case where people are using the flag, the right property, but then try to `require` an ESM file.

If this was meant as "you may want to add this flag", maybe something like this would be clearer:

```suggestion
    err.stack = 'Note: To load an ES module, please use --experimental-modules, ' +
```

Reply:
We shouldn’t be recommending people use a flag. Flags are for experimenting, never for production.

Reply:
In that case, perhaps let's only show this warning when the `--experimental-modules` flag is already in use.

Reply:
I believe I misunderstood the end result we want, 

```
if(!experimentalModules) {
 .... add the message "please use --experimental-modules"
}
```

something like this?

Reply:
The converse - only outputting the entire message within an `if (experimentalModules)` guard, and then removing the `--experimental-modules` reference in the message.